geotrellis/geotrellis-server
# Publish to Sonatype Nexus via CircleCI

## Overview

- Add GitHub pull request template
- Add GitHub issue template for releases
- Add a changelog, based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
- Update `./sbt` launcher
- Remove Travis, wire up CircleCI
- Remove Bintray, wire up Sonatype Nexus
- Set `version in ThisBuild` automatically using the output of `$ git describe`
- Replace `Makefile` with STRTA (See: [ADR 0000](https://github.com/azavea/architecture/blob/master/doc/arch/adr-0000-scripts-to-rule-them-all.md))
- Remove `sbt-microsites` plugin

Resolves #134 

### Checklist

- [x] `develop` branch is wired up, set to repository default, and branch protection is enabled
- [ ] This PR is merged to `develop`

### Notes

I see the removal of `sbt-microsites` as a possible point of contention. Here is why I removed it:

- I couldn't get `make serveDocs` to work (it just hangs up)
- I couldn't find the URL to the GitHub pages website advertised. I had to get it from repository settings.
- The site is a reflection of the README and has CSS errors
- The docs 404 (https://geotrellis.github.io/gtserver/latest/api)

I'm also open to leaving this functionality in, if we find `sbt-microsites` valuable, and would like to create issues to fix things.

## Testing Instructions

Try running STRTA on your workstation, for example:

```bash
$ ./scripts/test
...
[success] Total time: 16 s, completed Jul 22, 2019 4:11:00 PM
```

See passing CI checks below and [artifacts](https://oss.sonatype.org/content/repositories/snapshots/com/azavea/geotrellis/geotrellis-server-core_2.11/) on Sonatype Nexus.
